app/src/main/java/com/android/sample/ui/map/Map.kt

```diff
@@ -131,4 +155,119 @@ fun MapScreen(
             tabList = LIST_TOP_LEVEL_DESTINATION,
             selectedItem = Route.MAP)
       })
+
+  if (showBottomSheet) {
+    ModalBottomSheet(
+        onDismissRequest = { showBottomSheet = false },
+    ) {
+      Column(modifier = Modifier.fillMaxWidth().padding(16.dp)) {
+        // Ligne pour la croix en haut à droite
+        Row(
+            modifier = Modifier.fillMaxWidth().padding(8.dp),
+            horizontalArrangement = Arrangement.SpaceBetween, // Espace entre les éléments
+            verticalAlignment = Alignment.CenterVertically) {
+              SeeMoreDetailsButton(navigationActions)
+
+              // Icône de fermeture (croix) à droite
+              IconButton(onClick = { showBottomSheet = false }) {
+                Icon(
+                    imageVector = Icons.Default.Close,
+                    contentDescription = "Close",
+                    tint = MaterialTheme.colorScheme.onSurface)
+              }
+            }
+        if (selectedActivity != null) {
+          DisplayActivity(activity = selectedActivity!!)
```

Comment:
the "!!" is known to be unsafe, you could try something more concise as follows: 
selectedActivity?.let { activity ->
    DisplayActivity(activity = activity) }

Reply:
You are right, thank you!